### PR TITLE
Get system-probe profiles for flare over unix socket

### PR DESCRIPTION
**Pull request by hmahmood**
### What does this PR do?

1. Removes the requirement of setting `system_probe_config.debug_port` to get profiles from `system-probe` included in an agent flare
2. Makes available pprof handlers over `system-probe`'s unix socket
3. `system-probe` profiles are now retrieved over unix socket and included in the flare if `system_probe_config.enabled` is `true`

### Motivation

In flares received from customers, we would frequently see the `system-probe` profiles missing since the customer has not set `system_probe_config.debug_port` in the config; we would then have to ask the customer for another flare with this config set. This PR ensures the customer does not have to do any additional config changes to include `system-probe` profiles in their flare.

### Describe how to test/QA your changes

1. Ensure that system-probe is running. You can do this by enabling NPM: https://docs.datadoghq.com/network_monitoring/performance/setup/?tab=agentlinux#setup
2. run `agent flare --profile --local --email foo`
3. depending on which processes are running, there should be no errors getting profiles from that process. e.g., if `process-agent` is running, there should be no errors getting profiles for it. Same for `system-probe`
4. Say `N` to uploading the flare
5. Unzip the flare created (typically the flare zip file is in `/tmp`)
6. Verify that under the `profiles` folder you see the following files: ```system-probe-1st-heap.pprof  system-probe-block.pprof  system-probe-mutex.pprof
system-probe-2nd-heap.pprof  system-probe-cpu.pprof```

### CODE REVIEW

**Review by brycekahle — APPROVED** [review]

**Review by jeremy-hanna — APPROVED** [review]
👍 question re: connection clean up but approving so you don't need another review from us

File: cmd/agent/subcommands/flare/command.go
Comment (hmahmood) [reply cmd/agent/subcommands/flare/command.go]: I didn't write the original code, but my guess here would be that the flare command is going to exit soon anyway, so the connections should get closed.

I also updated this code to use the `RemoteSystemProbeUtil` object since that handles the windows case transparently. That object does not do closes either I believe since it is a singleton.

**Review by akarpz — APPROVED** [review]

**Pull request merged**
